pkg/operator/ceph/csi/csi.go

```diff
@@ -180,6 +180,11 @@ func (r *ReconcileCSI) setParams(ver *version.Info) error {
 		CSIParam.EnableCSIEncryption = true
 	}
 
+	CSIParam.CSIEnableMetadata = false
+	if strings.EqualFold(k8sutil.GetValue(r.opConfig.Parameters, "CSI_ENABLE_METADATA", "false"), "true") {
```

Comment:
Do we really need a setting for enabling metadata? Why wouldn't we always want to set the metadata on the volumes? Could we always enable metadata and avoid the setting?

Reply:
The reason why we are adding metadata on the backend images/subvolumes is to help monitor tools (like Prometheus) that might not have the access to 'Kubectl/oc' get commands, can list the images/subvolumes and try to get the metadata via their own implementation of python API (via librbd) and display the details in their dashboard. This will be helpful for close monitoring of the relationship between PV/PVC's and backend images/subvolumes.

1. The reason why keep `--setmetadata=false` by default is that maybe not all users are interested in getting these details. If that is the case making metadata set/get/rm calls would be a waste of few resources.
2. We are also hitting an RBD bug sometimes when playing around with metadata, see: https://github.com/ceph/ceph-csi/issues/2327 and https://bugzilla.redhat.com/show_bug.cgi?id=1989527

Reply:
Could you add a sentence in the comments around the setting in operator.yaml and the helm settings about this? To mention it's an optimization and perhaps even mention it's an experimental feature if there are issues with it.

Reply:
I have mentioned details in operator and helm settings about why this is set to false by default.

The issue is not directly related to metadata, hence I think its not worth to mention about it.